torch/_dynamo/variables/user_defined.py

```diff
@@ -1301,6 +1301,40 @@ def __init__(self, value, fields=None, **kwargs) -> None:
             fields = {}
         self.fields = fields
 
+    def reconstruct_to_python_object(self):
+        # TODO hack to enable experimenting on mark_traceable (need this to
+        # reconstruct input spec)
+        #
+        # Alternatives:
+        # 1. use `PyCodegen` to generate the bytecode, and invoke the function
+        #    to reconstruct the python objects.
```

Comment:
PyCodegen seems like the most legit way to do this yes

Reply:
Agree, but I'm debating a bit -- the codegen doesn't feel like an easily reusable component for our purposes here.

Luckily we probably only need `reconstruct_to_python_object` for `TreeSpec`, which should be a much narrower space to support, and maybe we can get away with this `as_python_constant`-flavor impl. I'll think more.